Sources/RTVIClientIOS/RTVIClient.swift

```diff
@@ -89,11 +97,18 @@ open class VoiceClient {
         }
     }
     
-    public init(baseUrl:String, transport:Transport, options: VoiceClientOptions) {
-        self.baseUrl = baseUrl
+    public init(baseUrl:String? = nil, transport:Transport, options: RTVIClientOptions) {
+        self.baseUrl = baseUrl ?? options.params.baseUrl
         self.options = options
         self.transport = transport
-        self.messageDispatcher = MessageDispatcher.init(transport: transport)
+        
+        let headers = options.customHeaders ?? options.params.headers
```

Comment:
On Android we combine both (same with the `requestData`) -- it could be confusing if setting one causes the other to be silently thrown away.